### PR TITLE
Check for unused WarpX environment variables when compiling

### PR DESCRIPTION
**Pull request by EtherealMC-Bit**
Currently if a user compiles using an environment variable starting with `WarpX` that is not picked up in `setup.py` they will have no way of knowing that the variable will not have any impact on their WarpX build. This PR adds the functionality to print out a warning if the user compiles with an environment variable that is not picked up in `setup.py`.

### CODE REVIEW

**Comment by peterscherpelz** [comment]
@MKieburtz I'm not sure this does quite what we want at present: A big use case for this was that we had a new CMake variable, but since setup.py needs to pass those variables to CMake, if the user forgot to add the line in eg line 182 of setup.py and below, the variable would be silently dropped going from python to cmake.

A possible way around is, like earlier in setup.py, set `env = os.environ.copy()` before line 180. Then change all the `.get` to `.pop`, and then after all those (line 209ish) check there are no more variables in `env` that start with WarpX (or better still `key.lower().startswith('warpx')` as the check, so capitalization errors are also caught).

File: setup.py [hunk 1]
```diff
@@ -174,34 +174,40 @@ def build_extension(self, ext):
...
+WarpX_amrex_src = env.pop('WarpX_amrex_src', '')
+WarpX_amrex_internal = env.pop('WarpX_amrex_internal', 'ON')
+WarpX_openpmd_src = env.pop('WarpX_openpmd_src', '')
+WarpX_openpmd_internal = env.pop('WarpX_openpmd_internal', 'ON')
+WarpX_picsar_src = env.pop('WarpX_picsar_src', '')
+WarpX_picsar_internal = env.pop('WarpX_picsar_internal', 'ON')
+
+for key in env.keys():
+    if key.lower().startswith('warpx'):
+        print(f"\nWARNING: compiled with '{key}', which is not a recognized argument\n")
```
Comment (ax3l) [inline setup.py]: Maybe slightly more precise:
```suggestion
        print(f"\nWARNING: Found environment variable '{key}', which is not a recognized WarpX option\n")
```

**Review by ax3l — COMMENTED** [review]
That's a great idea, thank you!
Just a minor suggestion added about the warning text.

**Review by ax3l — APPROVED** [review]
Thanks a lot, great idea! :)

**Pull request merged**